File: DailyChallenge/Day4/Day4.js
// DailyChallenge/Day4/Day4.js
        function calculateDaysTillJan25() {
            const today = new Date();
            const currentYear = today.getFullYear();
            const jan25ThisYear = new Date(currentYear, 0, 25); // January is month 0

            let targetDate;
            if (today > jan25ThisYear) {
                targetDate = new Date(currentYear + 1, 0, 25);
            } else {
                targetDate = jan25ThisYear;
            }

            const timeDiff = targetDate - today;
            const daysDiff = Math.ceil(timeDiff / (1000 * 60 * 60 * 24));

            document.getElementById('daysCount').innerText = `There are ${daysDiff} days till January 25.`;
        }

        window.onload = calculateDaysTillJan25;

File: DailyChallenge/Day4/Day4.py
#Days till January 25 python code

def CalculateDaysTillJan25():
    from datetime import datetime, timedelta

    today = datetime.now()
    current_year = today.year
    jan_25_this_year = datetime(current_year, 1, 25)

    if today > jan_25_this_year:
        target_date = datetime(current_year + 1, 1, 25)
    else:
        target_date = jan_25_this_year

    days_diff = (target_date - today).days
    print(f"There are {days_diff} days till January 25.")

CalculateDaysTillJan25()